src/components/implementation/fork_tests/voter/voter.c

```diff
@@ -176,7 +182,7 @@ void monitor(void) {
 			matches[j]++;						// for our own match. Is this how voting works? Do we even know anymore...
 			for (k = j + 1; k < snd_comp->nreplicas; k++) {
 				/* compare j to k  */
-				if (memcmp(snd_comp->replicas[j].buf_write, snd_comp->replicas[k].buf_write, 1) == 0) {
+				if (snd_comp->replicas[j].sz_write == snd_comp->replicas[k].sz_write && memcmp(snd_comp->replicas[j].buf_write, snd_comp->replicas[k].buf_write, snd_comp->replicas[j].sz_write) == 0) {
 					matches[j]++;
```

Comment:
is `matches` global data?  If so, that is broken.

Shouldn't `matches` just be in the `nmod_comp` structure, not wherever this is?

Shouldn't all of this code be gone (I'm assuming it is in the periodic thread which may be incorrect).